### DISCUSSION TITLE
Dashed line support

### DISCUSSION THREAD

**Discussion by hhpettersen** [patrykandpatrick/vico#469] [Ideas]
Is there any plans to introduce support for dashed line charts in any upcoming updates?
![Screenshot43](https://github.com/patrykandpatrick/vico/assets/52321453/a44ecb41-4cf4-4b82-95ea-76bc5e82e665)

**Comment by aaaleem** [comment]


**Reply by hhpettersen** [reply]
@aaaleem example posted in the description now.

**Reply by aaaleem** [reply]
@hhpettersen - try customizing `lineSpec` in  `lineChart`

**Comment by patrickmichalik** [comment]
Hello! We’ll be looking into adding such a feature.

**Reply by woxblom** [reply]
I also very much would like a dashed styling option of lines

**Reply by anshu7vyas** [reply]
Hello @patrickmichalik, I'm wondering if there will be support for dashed lines soon or, in the meantime if there's a way to customize `LineCartesianLayer`?

**Reply by patrickmichalik** [reply]
Hello, @anshu7vyas! I’m afraid we’re unable to share timelines for changes, as Vico is a free and open-source project that we develop in our spare time. `LineCartesianLayer` and `LineCartesianLayer.LineSpec` are open for extension—you can implement dashed lines by creating subclasses and making the required overrides.

**Reply by anshu7vyas** [reply]
Hey @patrickmichalik, thanks for your response. I understand, I created a PR which adds support for this by exposing `android.graphics.PathEffect` with the help of `LineSpec`. I'd appreciate your review on this if you think it's the correct way to add support. 
- #677 

Thank you for creating Vico, it's excellent, with extensive APIs and documentation. I appreciate the effort you've put into it. 🙌

**Reply by patrickmichalik** [reply]
Thanks for the kind words and the PR, @anshu7vyas! I’ll take a look shortly.

**Comment by Gowsky** [comment]
Hello, @hhpettersen. Vico [v2.0.0-beta.4](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-beta.4) adds the support for dashed lines.

**Comment by Gowsky** [comment]
[Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
